### PR TITLE
[MID-168] add animation to district filters and event filters

### PR DESCRIPTION
**Pull request by IvanKoshan**
Pr for this [task](https://trello.com/c/XBZ8O42I/168)

live server
https://ivan-koshans-mayak.vercel.app/

### CODE REVIEW

**Comment by ZakharovMaxim** [comment]
@IvanKoshan Can you also add some opacity/scale animation for the icon?

**Comment by IvanKoshan** [comment]
@ZakharovMaxim Added opacity and scale transition for the icon

**Comment by watchlar2000** [comment]
On small screens, if you scroll up to the end of the districts list and hover over the last button, it will expand and part of it will be out of the viewport.

Another thing is that on the Events page, there is the same list except it contains months. Should we also use slider on small screens there? Because now on the small screens the buttons just 'flex-wrap' to the next line when not fitting in one line. Or maybe we should go with this approach and have disctricts list buttons 'flex-wrap' instead of using slider. 

image

image

File: src/app/_components/MainPageSections/DistrictList.js [hunk 1]
```diff
@@ -7,16 +7,18 @@ import { CheckMark, Search } from '@icons';
...
-const commonIconStyle = 'h-4 w-4 transition-all';
 
 export function DistrictList({ list, className }) {
   const [selected, setSelected] = useState(0);
   const handleClick = index => {
     setSelected(index);
   };
 
+  useBodyScrollLock(true, 'x');
+
```
Comment (ZakharovMaxim) [inline src/app/_components/MainPageSections/DistrictList.js]: Isn't x axis always locked? I think we should have html {overflow-x: hidden;} styles globally

Comment (IvanKoshan) [reply src/app/_components/MainPageSections/DistrictList.js]: Yeah, that makes a lot of sense. Changed it

**Review by alex-popov-tech — APPROVED** [review]

**Comment by IvanKoshan** [comment]
@watchlar2000 Hi, as for the hover bug, this is true, it does move out of the viewport on hover. However I do think that this is not a big issue, considering that hover state isn't really all that used on mobile (users mostly just click on it). And fixing this doesn't seem so trivial, so it could take a lot of time. Seems like It'll be a separate bug task. I did change the event months filter to use the Swiper component.

**Review by watchlar2000 — APPROVED** [review]

**Review by ZakharovMaxim — APPROVED** [review]

**Pull request merged**